### ISSUE TITLE
IIS Detailed Error - 500.1001 - Internal Server Error

### ISSUE THREAD

**Issue by awatson1978** [fractal-code/meteor-azure#68]
Hi, digging into the issue more, we're receiving a `500.1001` error, which seems related to the web.config file.  Besides the NTFS errors and DDL permissions, which makes me wonder if the container is even built correctly, the thing that catches my eye about this error is the line referring to `https://symptomatic-chatbot:80/main.js`.  That doesn't seem to have a properly resolved URL, and I would have thought it would resolve to `https://symptomatic-chatbot.azurewebsites.net:80/main.js`, or something similar.  

Also, the command we're using to generate the build is the following:  
```
meteor-azure --debug --settings packages/biologio-curebot/config/settings.curebot.json --web-config ./web.config --architecture 64
```
It seems it's this change with the `--architecture` flag again.

#### Operating system  
macOS Mojave
Version 10.14.2

#### Meteor version  
Meteor 1.8

#### Package version (run `meteor-azure -v`)
v2.1.1

#### Command output with debug mode enabled
```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"> 
<html xmlns="http://www.w3.org/1999/xhtml"> 
<head> 
<title>IIS Detailed Error - 500.1001 - Internal Server Error</title> 

</head> 
<body> 
<div id="content"> 
<div class="content-container"> 
<h3>HTTP Error 500.1001 - Internal Server Error</h3> 
<h4>The page cannot be displayed because an internal server error has occurred.</h4> 
</div> 
<div class="content-container"> 
<fieldset><h4>Most likely causes:</h4> 
<ul> 	<li>IIS received the request; however, an internal error occurred during the processing of the request. The root cause of this error depends on which module handles the request and what was happening in the worker process when this error occurred.</li> 	<li>IIS was not able to access the web.config file for the Web site or application. This can occur if the NTFS permissions are set incorrectly.</li> 	<li>IIS was not able to process configuration for the Web site or application.</li> 	<li>The authenticated user does not have permission to use this DLL.</li> 	<li>The request is mapped to a managed handler but the .NET Extensibility Feature is not installed.</li> </ul> 
</fieldset> 
</div> 
<div class="content-container"> 
<fieldset><h4>Things you can try:</h4> 
<ul> 	<li>Ensure that the NTFS permissions for the web.config file are correct and allow access to the Web server's machine account.</li> 	<li>Check the event logs to see if any additional information was logged.</li> 	<li>Verify the permissions for the DLL.</li> 	<li>Install the .NET Extensibility feature if the request is mapped to a managed handler.</li> 	<li>Create a tracing rule to track failed requests for this HTTP status code. For more information about creating a tracing rule for failed requests, click <a href="http://go.microsoft.com/fwlink/?LinkID=66439">here</a>. </li> </ul> 
</fieldset> 
</div> 
 
<div class="content-container"> 
<fieldset><h4>Detailed Error Information:</h4> 
<div id="details-left"> 
<table border="0" cellpadding="0" cellspacing="0"> 
<tr class="alt"><th>Module</th><td>&nbsp;&nbsp;&nbsp;iisnode</td></tr> 
<tr><th>Notification</th><td>&nbsp;&nbsp;&nbsp;ExecuteRequestHandler</td></tr> 
<tr class="alt"><th>Handler</th><td>&nbsp;&nbsp;&nbsp;iisnode</td></tr> 
<tr><th>Error Code</th><td>&nbsp;&nbsp;&nbsp;0x00000002</td></tr> 
 
</table> 
</div> 
<div id="details-right"> 
<table border="0" cellpadding="0" cellspacing="0"> 
<tr class="alt"><th>Requested URL</th><td>&nbsp;&nbsp;&nbsp;https://symptomatic-chatbot:80/main.js</td></tr> 
<tr><th>Physical Path</th><td>&nbsp;&nbsp;&nbsp;D:\home\site\wwwroot\main.js</td></tr> 
<tr class="alt"><th>Logon Method</th><td>&nbsp;&nbsp;&nbsp;Anonymous</td></tr> 
<tr><th>Logon User</th><td>&nbsp;&nbsp;&nbsp;Anonymous</td></tr> 
 
</table> 
<div class="clear"></div> 
</div> 
</fieldset> 
</div> 
 
<div class="content-container"> 
<fieldset><h4>More Information:</h4> 
This error means that there was a problem while processing the request. The request was received by the Web server, but during processing a fatal error occurred, causing the 500 error. 
<p><a href="http://go.microsoft.com/fwlink/?LinkID=62293&amp;IIS70Error=500,1001,0x00000002,14393">View more information &raquo;</a></p> 
<p>Microsoft Knowledge Base Articles:</p> 
  
 
</fieldset> 
</div> 
</div> 
</body> 
</html> 
```

**Comment by dpdonohue** [comment]
+1 
there is a business need for this to get fixed, and additional resources available if needed
See also #67

**Comment by ramijarrar** [comment]
Strange - I recall seeing similar errors a while back due to issues with the preload mechanism (which have since been patched by the App Service team).

Can you confirm the details of your Kudu environment e.g:

<img width="553" alt="Screen Shot 2019-04-22 at 11 24 20 am" src="https://user-images.githubusercontent.com/9661116/56478267-493ad880-64f1-11e9-985c-644331fe0d53.png">

Also - please double check your application settings in the Azure portal are configured with Platform set to "64 Bit", HTTP set to "1.1", and "Always On/Web Sockets/ARR Affinity" all set to "On".

**Comment by ramijarrar** [comment]
Further to my comment above:

- Have you made any changes to your web config or are you using one of the samples from https://github.com/fractal-code/meteor-azure-web-config?

- Can you check for any errors in your application logs at Log Files > Application > *-stderr-*

**Comment by ramijarrar** [comment]
@dpdonohue We do provide consulting services if you need direct assistance with your deployment. Often times the issues are occurring upstream in package internals that need to be consolidated (this is something we have extensive experience with). Shoot me an email to rami@fractal.site if you want to discuss further.

**Comment by awatson1978** [comment]
![Screen Shot 2019-04-22 at 9 37 28 AM](https://user-images.githubusercontent.com/675910/56506513-556c6900-64e4-11e9-8e62-bcae32fc82a6.png)

![Screen Shot 2019-04-22 at 9 53 02 AM](https://user-images.githubusercontent.com/675910/56506587-99f80480-64e4-11e9-9eb9-ad5e6fa5766c.png)

Hi, we had attempted to make some changes to web.config a few months ago, but have since reverted to the default.config.

**Comment by ramijarrar** [comment]
Did you find any errors in your application logs in "Log Files > Application"?

**Comment by awatson1978** [comment]
Just more of the original error.

![Screen Shot 2019-04-23 at 7 52 40 PM](https://user-images.githubusercontent.com/675910/56625315-58f61200-6602-11e9-9e2c-c4a231d7a7dd.png)

**Comment by ramijarrar** [comment]
Those are the IIS logs (which are very difficult to sift though and are not Node aware).

I suspect there is some useful diagnostics in the console output which can be viewed in Kudu under "LogFiles > Application":

<img width="1046" alt="Screen Shot 2019-04-24 at 11 19 14 am" src="https://user-images.githubusercontent.com/9661116/56625931-e1b49280-6682-11e9-8af6-36cd7ebdadb0.png">

I would start with the `logging-errors.txt` file which catches crash details, everything else is formatted nicely in the index.html which can be opened directly @ https://YOUR_SITENAME.scm.azurewebsites.net/api/vfs/LogFiles/Application/index.html

FYI - this logged is handled by [IISNode](https://github.com/Azure/iisnode) and can be configured in the web config. The full list of options are listed [here](https://github.com/Azure/iisnode/blob/master/src/samples/configuration/web.config) - note "loggingEnabled", "logDirectory" and "maxLogFiles".

**Comment by awatson1978** [comment]
Finally got back to a proper internet connection.  

![Screen Shot 2019-05-01 at 12 17 10 AM](https://user-images.githubusercontent.com/675910/57006555-528a0c00-6ba7-11e9-8a58-6c36d866f9a7.png)

The issue appears to to be the database connection name.  We're using DocumentDB and the connection string provided by Azure (which doesn't include the database name) in the MONGO_URL environment setting.  This was working successfully back in November.  We've tried injecting the database name 'admin' into the URL, and I'm currently trying 'meteor' and some other variations.

**Comment by ramijarrar** [comment]
I would recommend opening a ticket with Azure as well if you have a [support plan](https://azure.microsoft.com/en-au/support/plans/) (we have done this on numerous occasions even with auxiliary tools such as IISNode and gotten generally useful help from technicians that are in direct contact with the application teams).

Also, if this does not seem deployment related - can we close the issue?

**Comment by ramijarrar** [comment]
Oh sorry, I've mixed up DocumentDB with the Azure Cosmos DB - I don't think they will want to help you with the AWS competitor :)

If you can connect to it with a mongo shell, it should be fairly easy to see where the data is stored. I doubt it would be 'meteor' as you would have to set that explicitly, seems more likely that AWS had a default which they removed to avoid this sort of confusion (wouldn't be admin as that is used for low level mongo stuff).

**Comment by awatson1978** [comment]
Well, I removed half of the packages in our app, and may have gotten it deployed again.  No errors in the `logging-errors.txt` and the Meteor framework is launching correctly (despite complaints about CosmoDB).  We're still getting an HTTP ERROR 500 from our outbound server, despite specifying the `web.config` file.  Has that been updated or changed lately?  

tl;dr - I rolled our code back to a working state, but things seem firewalled.

**Comment by awatson1978** [comment]
Have you seen this error before?

```
2019-05-04T05:15:40
System.ApplicationException: The trace listener AzureBlobTraceListener is disabled. ---> System.InvalidOperationException: The SAS URL for the cloud storage account is not specified. Use the environment variable 'DIAGNOSTICS_AZUREBLOBCONTAINERSASURL' to define it.
   at Microsoft.WindowsAzure.WebSites.Diagnostics.AzureBlobTraceListener.RefreshConfig()
   --- End of inner exception stack trace ---

2019-05-04T05:15:40
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.WindowsAzure.WebSites.Diagnostics.AzureBlobTraceListener.CleanupOldBlobLogs(Object state)
```

Received it while trying to deploy a web.config last night with HTTPS redirect.  But I've seen it elsewhere before.

**Comment by ramijarrar** [comment]
That error is nothing to worry about - you've enabled logging to blob container but have not configured any target for them. You can either disable it or setup with Azure storage:

<img width="377" alt="Screen Shot 2019-05-08 at 5 10 48 am" src="https://user-images.githubusercontent.com/9661116/57326471-eaf83180-714f-11e9-8ac1-ec0b904157c2.png">

**Comment by ramijarrar** [comment]
> We're still getting an HTTP ERROR 500 from our outbound server, despite specifying the web.config file. Has that been updated or changed lately?

This is common either during the deploy process (unless you have configured a separate slot for zero downtime swap) or for a few minutes after while it boots up - can you post a screen grab?

**Comment by ramijarrar** [comment]
Closed due to inactivity.
